pkg/auth/user/required_attribute_checker_test.go

```diff
@@ -0,0 +1,98 @@
+package user
+
+import (
+	"testing"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCheckRequiredAttributesImpl_Check(t *testing.T) {
+	cases := map[string]struct {
```

Comment:
also suggestions for tests:

1. `required attribute set but value does not match`
2. `required attribute in the map but empty array value`